### PR TITLE
Feature/6 followers and following

### PR DESCRIPTION
**Pull request by apatchenkov**

### CODE REVIEW

File: app/assets/javascripts/relationships.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (nickskalkin) [inline app/assets/javascripts/relationships.coffee]: If there is no need in this file - do not generate it. May be you should turn off the generator. Also, lets not use cofeescript.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class UsersController < ApplicationController
+  def show
+    @user = User.find(params[:id])
```
Comment (nickskalkin) [inline app/controllers/users_controller.rb]: Move it to helper method

```
private

def user
  @user ||= User.find(params[:id])
end
helper_method :user
```

Also, try not to use instance variables

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  def following
+    @title = 'Following'
+    @user  = User.find(params[:id])
```
Comment (nickskalkin) [inline app/controllers/users_controller.rb]: Use helper method, described above

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    @users = @user.following.paginate(page: params[:page])
+    render 'show_follow'
+  end
+
+  def followers
+    @title = 'Followers'
+    @user  = User.find(params[:id])
```
Comment (nickskalkin) [inline app/controllers/users_controller.rb]: save

File: app/helpers/relationships_helper.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+module RelationshipsHelper
```
Comment (nickskalkin) [inline app/helpers/relationships_helper.rb]: Don't generate it when it's not needed

File: app/models/user.rb [hunk 6]
```diff
@@ -1,6 +1,29 @@
...
+                                   foreign_key: 'followed_id',
+                                   dependent:   :destroy
+  has_many :following, through: :active_relationships,  source: :followed
+  has_many :followers, through: :passive_relationships, source: :follower
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+  # Follows a user.
```
Comment (nickskalkin) [inline app/models/user.rb]: Do we need this comment? :)

File: app/views/shared/_stats.html.haml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+- @user ||= current_user
+.stats
+  %a{:href => "#{following_user_path(@user)}"}
+    %strong#following.stat
+      = @user.following.count
+    following
+  %a{:href => "#{followers_user_path(@user)}"}
+    %strong#followers.stat
+      = @user.followers.count
+    followers
```
Comment (nickskalkin) [inline app/views/shared/_stats.html.haml]: Use `t` helper and save `followers` to locales

File: app/views/shared/_stats.html.haml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+- @user ||= current_user
+.stats
+  %a{:href => "#{following_user_path(@user)}"}
+    %strong#following.stat
+      = @user.following.count
+    following
```
Comment (nickskalkin) [inline app/views/shared/_stats.html.haml]: Use `t` helper and save `following` to locales

File: app/views/static_pages/home.html.haml [hunk 9]
```diff
@@ -0,0 +1,9 @@
+%h1 This is Bithub home page.
```
Comment (nickskalkin) [inline app/views/static_pages/home.html.haml]: Save all texts in locales :)

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (karpinovsky) [reply app/controllers/users_controller.rb]: ^__^

File: app/controllers/users_controller.rb [hunk 10]
```diff
@@ -0,0 +1,35 @@
+class UsersController < ApplicationController
+  helper_method :user, :users_paginate_followers, :users_paginate_following
+
+  def show
+    user
```
Comment (nickskalkin) [inline app/controllers/users_controller.rb]: there is no need in method `user`. Just leave it empty

```
def show; end
```

File: app/controllers/users_controller.rb [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+  end
+
+  private
+
+  def user
+    @user ||= User.find(params[:id])
+  end
+
+  def users_paginate_followers
+    @users = @user.followers.paginate(page: params[:page])
```
Comment (nickskalkin) [inline app/controllers/users_controller.rb]: `user` instead of `@user`

File: app/controllers/users_controller.rb [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+  end
+
+  def users_paginate_following
+    @users = @user.following.paginate(page: params[:page])
```
Comment (nickskalkin) [inline app/controllers/users_controller.rb]: same

File: app/views/static_pages/home.html.haml [hunk 13]
```diff
@@ -0,0 +1,9 @@
+%h1= t('This is Bithub home page.')
```
Comment (nickskalkin) [inline app/views/static_pages/home.html.haml]: it should be like `= t('.title')` and in config/locales/en.yml

```
en:
  static_pages:
    home:
      title: 'This is Bithub home page'
```

**Pull request merged**